packages/web/src/common/store/upload/sagas.js

```diff
@@ -1156,6 +1161,35 @@ function* uploadMultipleTracks(tracks) {
   yield put(cacheActions.setExpired(Kind.USERS, account.user_id))
 }
 
+function* processTracksForUpload(tracks) {
+  const getFeatureEnabled = yield getContext('getFeatureEnabled')
+  const isUsdcPurchaseEnabled = yield call(
+    getFeatureEnabled,
+    FeatureFlags.USDC_PURCHASES
+  )
+  if (!isUsdcPurchaseEnabled) return
+
+  const ownerAccount = yield select(getAccountUser)
+  const wallet = ownerAccount.erc_wallet ?? ownerAccount.wallet
+  const ownerUserbank = yield getUSDCUserBank(wallet)
+
+  tracks.forEach((track) => {
```

Comment:
Nice, is this section something we'd want to double check with usdc folks on requirements or you already did. And so I'm clear this would actually be what we'd use for v2 upload too?